JamfSync/Model/Jcds2Dp.swift

```diff
@@ -13,12 +13,19 @@ class Jcds2Dp: DistributionPoint {
     var urlSession: URLSession?
     var downloadTask: URLSessionDownloadTask?
     var dispatchGroup: DispatchGroup?
+    
+    // for aws uri encoding - BIG-RAT
+    var myCharacterSet = CharacterSet()
 
     init(jamfProInstanceId: UUID? = nil, jamfProInstanceName: String? = nil) {
         super.init(name: "JCDS")
         self.jamfProInstanceId = jamfProInstanceId
         self.jamfProInstanceName = jamfProInstanceName
         self.willDownloadFiles = true
+        
+        // character set for asw header uri encoding - BIG-RAT
+        myCharacterSet.formUnion(.alphanumerics)
+        myCharacterSet.insert(charactersIn: "/-._~")
```

Comment:
Actually, I have this same set called allowedUrlCharacters in the awsSignatureV4 function. So maybe just move that to here and use the same one in both places.